app/scripts/directives/nav.js

```diff
@@ -273,10 +291,47 @@ angular.module('openshiftConsole')
         };
 
         $scope.catalogLandingPageEnabled = !Constants.DISABLE_SERVICE_CATALOG_LANDING_PAGE;
-        var select = $elem.find('.selectpicker');
-        var options = [];
 
-        var updateOptions = function() {
+        var contextSelector = $elem.find('.contextselector');
+        var previousContextSelectorValue;
+        $scope.clusterConsoleURL = window.OPENSHIFT_CONSTANTS.TECTONIC_URL;
+        contextSelector
+          .on('loaded.bs.select', function () {
+            setContextPickerSelectedOption(contextSelector);
+          })
+          .on('shown.bs.select', function() {
+            previousContextSelectorValue = $(this).val();
+          })
+          .change(function() {
+            // // unset focus on .dropdown-toggle so outline isn't present
+            // $('.contextselector .dropdown-toggle').blur();
+            var val = $(this).val();
+            switch (val) {
+              case "catalog":
+               goToURL(catalogPath);
+               break;
+              case "application-console":
+               goToURL(projectsPath);
+               break;
+              case "cluster-console":
+                window.location.assign($scope.clusterConsoleURL);
+                break;
+              case "set-home-page":
```

Comment:
I'd take the set-home-page stuff out for now since it's unclear when we will add that (both here and in the view)